modules/core/04-channel/keeper/upgrade.go

```diff
@@ -131,125 +96,98 @@ func (k Keeper) ChanUpgradeTry(ctx sdk.Context, portID string, channelID string,
-		channel.Counterparty.ChannelId, counterpartyUpgradeSequence); err != nil {
-		return 0, err
-	}
+	/*
+		replace the following with:
+
+		selfHeight := clienttypes.GetSelfHeight(ctx)
+		selfTime := uint64(ctx.BlockTime.UnixNano())
+		if info, hasPassed := counterpartyProposedUpgrade.Timeout.Status(selfHeight, selfTime); hasPassed {
+			return 0, types.NewErrorReceipt(upgradeSequence, info)
+		}
+	*/
 
 	// check if upgrade timed out by comparing it with the latest height of the chain
 	selfHeight := clienttypes.GetSelfHeight(ctx)
+	timeoutHeight := counterpartyProposedUpgrade.Timeout.TimeoutHeight
 	if !timeoutHeight.IsZero() && selfHeight.GTE(timeoutHeight) {
-		if err := k.RestoreChannelAndWriteErrorReceipt(ctx, portID, channelID, upgradeSequence, types.ErrUpgradeTimeout); err != nil {
+		if err := k.WriteErrorReceipt(ctx, portID, channelID, upgradeSequence, types.ErrUpgradeTimeout); err != nil {
 			return 0, errorsmod.Wrap(types.ErrUpgradeAborted, err.Error())
 		}
 		return 0, errorsmod.Wrapf(types.ErrUpgradeAborted, "block height >= upgrade timeout height (%s >= %s)", selfHeight, timeoutHeight)
 	}
 
 	// check if upgrade timed out by comparing it with the latest timestamp of the chain
+	timeoutTimestamp := counterpartyProposedUpgrade.Timeout.TimeoutTimestamp
 	if timeoutTimestamp != 0 && uint64(ctx.BlockTime().UnixNano()) >= timeoutTimestamp {
 		upgradeSequence = uint64(0)
-		if err := k.RestoreChannelAndWriteErrorReceipt(ctx, portID, channelID, upgradeSequence, types.ErrUpgradeTimeout); err != nil {
+		if err := k.WriteErrorReceipt(ctx, portID, channelID, upgradeSequence, types.ErrUpgradeTimeout); err != nil {
 			return 0, errorsmod.Wrap(types.ErrUpgradeAborted, err.Error())
 		}
 		return 0, errorsmod.Wrapf(types.ErrUpgradeAborted, "block timestamp >= upgrade timeout timestamp (%s >= %s)", ctx.BlockTime(), time.Unix(0, int64(timeoutTimestamp)))
 	}
 
-	switch channel.State {
-	case types.OPEN:
-		upgradeSequence = uint64(0)
-		if seq, found := k.GetUpgradeSequence(ctx, portID, channelID); found {
-			upgradeSequence = seq
-		}
+	// increment upgrade sequence for new upgrade attempt
+	if channel.State == types.OPEN {
+		channel.UpgradeSequence = channel.UpgradeSequence + 1
 
 		// if the counterparty upgrade sequence is ahead then fast forward so both channel ends are using the same sequence for the current upgrade
-		if counterpartyUpgradeSequence > upgradeSequence {
-			upgradeSequence = counterpartyUpgradeSequence
-			k.SetUpgradeSequence(ctx, portID, channelID, upgradeSequence)
-		} else {
-			errorReceipt := types.NewErrorReceipt(upgradeSequence, errorsmod.Wrapf(types.ErrUpgradeAborted, "counterparty chain upgrade sequence <= upgrade sequence (%d <= %d)", counterpartyUpgradeSequence, upgradeSequence))
-			// the upgrade sequence is incremented so both sides start the next upgrade with a fresh sequence.
-			upgradeSequence++
```

Comment:
I have a question about this. Is it necessary to increment here? I feel like we should only increment the sequence when attempting an upgrade?